amethyst_renderer/src/hidden.rs

```diff
@@ -0,0 +1,22 @@
+use amethyst_core::specs::{
+    prelude::Component,
+    storage::{FlaggedStorage, NullStorage},
+};
+
+/// Hidden mesh component
+/// Useful for entities, that should not be rendered, but stay loaded in memory.
+#[derive(Clone, Debug, Default)]
+pub struct Hidden;
+
+impl Component for Hidden {
+    type Storage = NullStorage<Self>;
+}
+
+/// Like [Hidden](struct.Hidden.html), but can propagate through children when the [HideHierarchySystem](struct.HideHierarchySystem.html)
+/// is enabled in the [RenderBundle](struct.RenderBundle.html).
+#[derive(Clone, Debug, Default)]
+pub struct HiddenPropagate;
```

Comment:
Note: I do like the separation of having one that propagates and the other one who doesn't. :+1: